File: chaincode/lib/item.js
'use strict';

class Item {
    /**
     *
     * Item
     *
     * Constructor for an Item object.
     *  
     * @param itemId - unique Id which corresponds to a created Item
     * @param name - Item name
     * @param description - Item description
     * @param price - Item price
     * @param cancellation - true if cancellation is allowed otherwise false
     * @param rebooking - true if rebooking is allowed otherwise false
     * @param groupId - id of a group of items
     * @param providerId - id of an Item's provider
     * @returns - Item object
     */
    constructor(itemId, name, description, price, cancellation, rebooking, groupId, providerId) {

        this.itemId = itemId;
        this.name = name;
        this.description = description;
        this.price = price;
        this.cancellation = cancellation;
        this.rebooking = rebooking;
        this.groupId = groupId;
        this.providerId = providerId;
        this.type = 'item';
        if (this.__isContract) {
            delete this.__isContract;
        }
        return this;

    }

}
module.exports = Item;

File: reservation_system/application/lib/models/item.dart
class Item {
  final String id;
  final String name;
  final String description;
  final double price;
  final String cancellation;
  final String rebooking;
  final String providerId;
  final String? ownerId;

  Item(
    this.id,
    this.name,
    this.description,
    this.price,
    this.cancellation,
    this.rebooking,
    this.providerId,
    this.ownerId,
  );

  factory Item.fromJson(Map<String, dynamic> json) {
    return Item(
      json['itemId'],
      json['name'],
      json['description'],
      double.parse(json['price']),
      json['cancellation'],
      json['rebooking'],
      json['providerId'],
      json['ownerId'],
    );
  }

  Map toJson() => {
        'itemId': id,
        'name': name,
        'description': description,
        'price': price,
        'cancellation': cancellation,
        'rebooking': rebooking,
        'providerId': providerId,
        'ownerId': ownerId,
      };
}
